code.py

```diff
@@ -0,0 +1,311 @@
+    st.pyplot(fig2)
+if selected == "SUPPORT VECTOR REGRESSION":
+    st.title("STOCK ANALYSIS")
+    user_input = st.text_input("ENTER YOUR STOCK", 'NFLX')
+    df = web.DataReader(user_input, 'yahoo', start, end)
+
+    st.subheader("Data")
+    st.write(df.describe())
+
+    st.subheader("SAMPLE")
+    st.write("HEAD")
+    forecast = 365
+    df['Prediction'] = df[['Adj Close']].shift(-forecast)
+    st.write(df.head())
+
+    # tail
+    st.write("TAIL")
+    st.write(df.tail())
+    # visualization
+    st.subheader("CLOSING PRICE V/S TIME", 'r')
+    fig = plt.figure(figsize=(18, 6))
+    plt.plot(df.Close, 'g')
+    st.pyplot(fig)
+
+    st.subheader("100 DAYS MOVING AVERAGE", 'r')
+    ma100 = df.Close.rolling(100).mean()
+    fig = plt.figure(figsize=(18, 6))
+    plt.plot(ma100, 'r')
+    plt.plot(df.Close, 'g')
+    st.pyplot(fig)
+
+
+    # X
+    X = np.array(df.drop(['Prediction'], 1))
+    X = X[:-forecast]
+    # y
+    y = np.array(df['Prediction'])
+    y = y[:-forecast]
+
+    x_train, x_test, y_train, y_test = train_test_split(X, y, test_size=0.2)
+
+    svr_rbf = SVR(kernel='rbf', C=1e3, gamma=0.1)
+    svr_rbf.fit(x_train, y_train)
+
+    svm_confidence = svr_rbf.score(x_test, y_test)
+    st.write('SVM confidence: ', svm_confidence)
+
+    x_forecast_out = np.array(df.drop(['Prediction'], 1))[-forecast:]
+
+    svm_prediction = svr_rbf.predict(x_forecast_out)
+    st.write(svm_prediction)
+
+    st.subheader("PREDICTION GRAPH")
+    predictions = svm_prediction
+    valid = df[X.shape[0]:]
+    valid['Predictions'] = predictions
+    fig3 = plt.figure(figsize=(16, 8))
+    plt.title('SUPPORT VECTOR REGRESSION')
+    plt.xlabel('Days')
+    plt.ylabel('Closing Price')
+    plt.plot(df['Adj Close'])
+    plt.plot(valid[['Adj Close', 'Predictions']])
+    plt.legend(['Orignal', 'Val', 'Predicted'])
+    st.pyplot(fig3)
+if selected == 'RANDOM FOREST REGRESSION':
+    st.title("STOCK ANALYSIS")
+    user_input = st.text_input("ENTER YOUR STOCK", 'NFLX')
+    df = web.DataReader(user_input, 'yahoo', start, end)
+
+    st.subheader("Data")
+    st.write(df.describe())
+
+    st.write("HEAD")
+    forecast = 365
+    df['Prediction'] = df[['Adj Close']].shift(-forecast)
+    st.write(df.head())
+
+    # tail
+    st.write("TAIL")
+    st.write(df.tail())
+
+    st.subheader("CLOSING PRICE V/S TIME", 'r')
+    fig = plt.figure(figsize=(18, 6))
+    plt.plot(df.Close, 'g')
+    st.pyplot(fig)
+
+    st.subheader("100 DAYS MOVING AVERAGE", 'r')
+    ma100 = df.Close.rolling(100).mean()
+    fig = plt.figure(figsize=(18, 6))
+    plt.plot(ma100, 'r')
+    plt.plot(df.Close, 'g')
+    st.pyplot(fig)
+
+    future_days = 365
+    df['Prediction'] = df[['Close']].shift(-future_days)
+    df.tail(4)
+
+    #X
+    X = np.array(df.drop(['Prediction'], 1))[:-future_days]
+    #Y
+    y = np.array(df['Prediction'])[:-future_days]
+
+    x_train, x_test, y_train, y_test = train_test_split(X, y, test_size=0.25)
+
```

Comment:
You are using the variable X in valid = df[X.shape[0]:], but it's not defined anywhere in the given code. I guess you should use X_train.  And also the predictions array is being assigned to the valid DataFrame's 'Predictions' column through the line valid['Predictions'] = predictions. Assigning values to a new column in a slice may not function as intended because valid is produced as a slice of the original DataFrame df. You can make a new DataFrame just for the forecasts to get around this problem.